lib/timers.js

```diff
@@ -222,17 +241,21 @@ exports.clearTimeout = function(timer) {
 };
 
 
-exports.setInterval = function(callback, repeat) {
+exports.setInterval = function(callback, repeat, arg1, arg2, arg3) {
   repeat *= 1; // coalesce to number or NaN
 
   if (!(repeat >= 1 && repeat <= TIMEOUT_MAX)) {
     repeat = 1; // schedule on next tick, follows browser behaviour
   }
 
-  var timer = new Timeout(repeat);
-  var args = Array.prototype.slice.call(arguments, 2);
+  var timer = new Timeout(repeat), len = arguments.length - 2, args, i;
   timer._onTimeout = wrapper;
   timer._repeat = true;
+  if (len > 2) {
```

Comment:
Doesn't sound too good: the `switch` statement is inside a function that is executed multiple times. The initialization only needs to happen once. Clarified this with a comment.